File: sections/10_greedy_constructive/problems/c_bounded_sum_sequence/solution.cpp
#include <bits/stdc++.h>
using namespace std;

string construct(long long n, long long m, long long s) {
    if (s < n || s > n * m) return "IMPOSSIBLE";

    vector<long long> ans;
    ans.reserve((size_t)n);
    long long remaining = s;
    for (long long pos = 0; pos < n; ++pos) {
        long long left = n - pos - 1;
        long long x = max(1LL, remaining - left * m);
        ans.push_back(x);
        remaining -= x;
    }

    stringstream ss;
    for (int i = 0; i < (int)ans.size(); ++i) {
        if (i) ss << ' ';
        ss << ans[i];
    }
    return ss.str();
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int T;
    cin >> T;
    while (T--) {
        long long n, m, s;
        cin >> n >> m >> s;
        cout << construct(n, m, s) << '\n';
    }

    return 0;
}

File: sections/10_greedy_constructive/problems/c_bounded_sum_sequence/solution.py
"""Reference solution for C. Bounded Sum Sequence."""

import sys


def construct(n: int, m: int, s: int) -> str:
    if s < n or s > n * m:
        return "IMPOSSIBLE"
    ans: list[int] = []
    remaining = s
    for pos in range(n):
        left = n - pos - 1
        x = max(1, remaining - left * m)
        ans.append(x)
        remaining -= x
    return " ".join(map(str, ans))


def main() -> None:
    data = list(map(int, sys.stdin.buffer.read().split()))
    if not data:
        return
    t = data[0]
    idx = 1
    out: list[str] = []
    for _ in range(t):
        n = data[idx]
        m = data[idx + 1]
        s = data[idx + 2]
        idx += 3
        out.append(construct(n, m, s))
    sys.stdout.write("\n".join(out))


if __name__ == "__main__":
    main()

